app/facades/cocktail_facade.rb

```diff
@@ -31,26 +31,35 @@ def self.retrieve_cocktail(cocktail_id)
   end
 
   def self.get_ingredients(cocktail_data)
-    ingredients = []
-    cocktail_data.each do |key, value|
-      stringify_key = key.to_s
-      if stringify_key.include?("Ingredient")
-        value = "" if value.nil?
-        ingredients << value
+    cocktail_data.map do |key, value|
+      if key.to_s.include?("Ingredient")
+        value ||= ""
       end
-    end
-    ingredients
+    end.compact
   end
 
   def self.get_measurements(cocktail_data)
-    measurements = []
-    cocktail_data.each do |key, value|
-      stringify_key = key.to_s
-      if stringify_key.include?("Measure")
-        value = "" if value.nil?
-        measurements << value
+    cocktail_data.map do |key, value|
+      if key.to_s.include?("Measure")
+        value ||= ""
       end
+    end.compact
+  end
+
+  def self.retrieve_search_results(query, user_id)
+    response = CocktailService.search_cocktails(query)
+
+    response[:drinks].map do |drink|
+      {
+        id: drink[:idDrink],
+        name: drink[:strDrink],
+        thumbnail: drink[:strDrinkThumb],
+        rating: if Rating.find_by(cocktail_id: drink[:idDrink], user_id: user_id).present?
+                  Rating.find_by(cocktail_id: drink[:idDrink], user_id: user_id).stars
+                else
+                  0
+                end
```

Comment:
We should probably pull this into a separate method. That can be a refactor on another branch though, if needed. Not critical to MVP.

Reply:
Will do!